packages/dds/tree/src/simple-tree/discrepancies.ts

```diff
@@ -294,21 +443,42 @@ function* getFieldDiscrepancies(
 		} satisfies AllowedTypeDiscrepancy;
 	}
 
-	if (view.kind !== stored.kind) {
+	const viewKind =
+		convertFieldKind.get(view.kind) ??
+		fail("A conversion from a FieldKind to a FlexFieldKind should exist");
+
+	// This checks if the field kind in the view schema is not compatible with the stored schema.
+	// We cannot detect if the view schema is a sequence using the kind property so it is passed in separately.
+	// TODO: This is a temporary workaround until the comparison logic is redesigned.
+	if (
+		(viewKindIsSequence && stored.kind !== "Sequence") ||
+		(!viewKindIsSequence && viewKind.identifier !== stored.kind)
+	) {
 		yield {
 			identifier,
 			fieldKey,
 			mismatch: "fieldKind",
-			view: view.kind,
+			view: viewKind.identifier,
 			stored: stored.kind,
 		} satisfies FieldKindDiscrepancy;
 	}
 }
 
+/**
+ * Tracks discrepancies between a view schema and a stored schema for nodes that are treated as object nodes in the stored schema.
```

Comment:
I don't think "tracks" is the right verb here.

Objects which track something generally observe some value overt time, where this just computes them once and returns the result.
```suggestion
 * Tracks discrepancies between a view schema and a stored schema for nodes that are treated as object nodes in the stored schema.
```

```suggestion
 * Computes discrepancies between a view schema and a stored schema for nodes that are treated as object nodes in the stored schema.
```

Reply:
Also the function name should probably be updated, to maybe `objectNodeDiscrepancies` or `computeObjectNodeDiscrepancies`